packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -0,0 +1,1274 @@
+  ///
+  /// If not specified, [child] will be shown.
+  ///
+  /// The preview is often used to show a slight variation of the [child]. For
+  /// example, the child could be given rounded corners in the preview but have
+  /// sharp corners when in the page.
+  ///
+  /// In addition to the current [BuildContext], the function is also called
+  /// with an [Animation] and the [child]. The animation is used internally to
+  /// animate the CupertinoContextMenu open. It can be used to animate the
+  /// preview in sync with the opening of the CupertinoContextMenu. The child
+  /// parameter provides access to the child displayed when the
+  /// CupertinoContextMenu is closed.
+  ///
+  /// {@tool sample}
+  ///
+  /// Below is an example of using `previewBuilder` to show an image tile that's
+  /// similar to each tile in the iOS iPhoto app's context menu. Several of
+  /// these could be used in a GridView for a similar effect.
+
+  /// When opened, the child animates to show its full aspect ratio and has
+  /// rounded corners. The larger size of the open CupertinoContextMenu allows
+  /// the FittedBox to fit the entire image, even when it has a very tall or
+  /// wide aspect ratio compared to the square of a GridView, so this animates
+  /// into view as the CupertinoContextMenu is opened. The preview is swapped in
+  /// right when the open animation begins, which includes the rounded corners.
+  ///
+  /// ```dart
+  /// CupertinoContextMenu(
+  ///   child: FittedBox(
+  ///     fit: BoxFit.cover,
+  ///     child: Image.asset('assets/photo.jpg'),
+  ///   ),
+  ///   // The FittedBox in the preview here allows the image to animate its
+  ///   // aspect ratio when the CupertinoContextMenu is animating its preview
+  ///   // widget open and closed.
+  ///   previewBuilder: (BuildContext context, Animation<double> animation, Widget child) {
+  ///     return FittedBox(
+  ///       fit: BoxFit.cover,
+  ///       // This ClipRRect rounds the corners of the image when the
+  ///       // CupertinoContextMenu is open, even though it's not rounded when
+  ///       // it's closed. It uses the given animation to animate the corners
+  ///       // in sync with the opening animation.
+  ///       child: ClipRRect(
+  ///         borderRadius: BorderRadius.circular(64.0 * animation.value),
+  ///         child: Image.asset('assets/photo.jpg'),
+  ///       ),
+  ///     );
+  ///   },
+  ///   actions: <CupertinoContextMenuAction>[
+  ///     CupertinoContextMenuAction(
+  ///       child: const Text('Action one'),
+  ///       onPressed: () {},
+  ///     ),
+  ///   ],
+  /// ),
+  /// ```
+  ///
+  /// {@end-tool}
+  final PreviewBuilder previewBuilder;
+
+  @override
+  _CupertinoContextMenuState createState() => _CupertinoContextMenuState();
+}
+
+class _CupertinoContextMenuState extends State<CupertinoContextMenu> with TickerProviderStateMixin {
+  final GlobalKey _childGlobalKey = GlobalKey();
+  double _childOpacity = 1.0;
```

Comment:
I'm trying this out.

Actually @goderbauer I got this code from you awhile back.  I was asking about animating to a final laid out state, and you suggested rendering the first frame offstage and sent me this gist: https://gist.github.com/goderbauer/b136e871fac33960dbd539ea1930ff97

Any idea if there is an advantage to Offstage and TickerMode instead?

Reply:
Ah I think it's because PopupRoute is already using Offstage internally, and so this widget is just managing the `offstage` variable.

https://github.com/flutter/flutter/blob/ed36fe5ddde84ffd9b6380687bc6a2f37b0790ab/packages/flutter/lib/src/widgets/routes.dart#L663

Reply:
What I originally meant was to use `Offstage` + `TickerMode` to hide the original child instead of `Opacity`. Should we stop the animation in the original child when it's hidden, by adding a `TickerMode` parent? Toggling `Opacity` between 0 and 1 seems fine.

Reply:
Oh I see, sorry!  That makes more sense and is probably more performant.  I'll change it.

Reply:
Actually after changing it something broke.  I was getting the size of the Opacity widget using a GlobalKey, and I couldn't do that with an Offstage widget (even when it was onstage).  It gave me weird values and broke the tests.

My compromise is to use a TickerMode and an Opacity widget together, but let me know if you know why getting the size wasn't working.